### PR TITLE
Add integration tests for Bijectors

### PR DESCRIPTION
**Pull request by mhauru**
There have been some regressions that cause various crashes and errors with Bijectors.jl and Enzyme v0.13. To guard against this in the future, I think it would be useful to introduce some integration tests for Bijectors, like we are doing in #1813 and #1819 for Turing.jl and Distributions.jl.

I've added a basic check of running forward and reverse mode on all bijector types with at least one argument, plus some extra ones that have caused trouble in the past. I've marked all failing ones, and reported them as separate issues. Here's a list:
* https://github.com/EnzymeAD/Enzyme.jl/issues/2035
* https://github.com/EnzymeAD/Enzyme.jl/issues/2030
* https://github.com/EnzymeAD/Enzyme.jl/issues/2033
* https://github.com/EnzymeAD/Enzyme.jl/issues/2034
* https://github.com/EnzymeAD/Enzyme.jl/issues/2036
* https://github.com/EnzymeAD/Enzyme.jl/issues/2029
* https://github.com/EnzymeAD/Enzyme.jl/issues/2041

cc @yebai

### CODE REVIEW

File: test/integration/Bijectors/runtests.jl [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    r_mode = if (case.runtime_activity === Both || case.runtime_activity === Reverse)
+        Enzyme.set_runtime_activity(Enzyme.Reverse)
+    else
+        Enzyme.Reverse
+    end
+
+    if !(case.skip === Forward) && !(case.skip === Both)
+        if case.broken === Both || case.broken === Forward
+            @test_broken(
+                collect(Enzyme.gradient(f_mode, Enzyme.Const(f), x...)[1]) ≈ finitediff,
```
Comment (wsmoses) [inline test/integration/Bijectors/runtests.jl:77]: why are these collect's needed?

Comment (mhauru) [reply test/integration/Bijectors/runtests.jl]: They are not. Removed. Thanks for spotting.

**Comment by mhauru** [comment]
Had to mark one more test as broken, CI now passes. As Hong said, I won't have time to minimise the issues I opened in the near future. Would you be happy to merge as is? Should be better than nothing, and we can add in more tests as we close the individual issues.

**Comment by wsmoses** [comment]
I think the segfaults need further reductions at least, lest we not be able to mark them as broken

**Comment by mhauru** [comment]
I added back in a bunch of tests that were unblocked by the recent fixes. Now the only one marked with `skip=Reverse` is the one that hangs.

**Comment by mhauru** [comment]
There's now a new seg fault I'm unable to reproduce locally. @wsmoses (or anyone else), would you have a Linux machine to try it on? I'm on Mac, CI runs on Ubuntu 22.04.

File: test/integration/Bijectors/runtests.jl [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+end
+
+@testset "Bijectors integration tests" begin
+    test_cases = TestCase[
+        sum_b_binv_test_case(Bijectors.VecCorrBijector(), 3),
+        sum_b_binv_test_case(Bijectors.VecCorrBijector(), 0),
+        sum_b_binv_test_case(Bijectors.CorrBijector(), (3, 3)),
+        # TODO(mhauru) Skip Reverse because of
+        # https://github.com/EnzymeAD/Enzyme.jl/issues/2033
+        sum_b_binv_test_case(Bijectors.CorrBijector(), (0, 0); skip=Reverse),
```
Comment (wsmoses) [inline test/integration/Bijectors/runtests.jl:130]: this should no longer be skipped

File: test/integration/Bijectors/runtests.jl [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+        sum_b_binv_test_case(
+            Bijectors.Permute([
+                0 1 0;
+                1 0 0;
+                0 0 1
+            ]),
+            (3, 3),
+        ),
+        # TODO(mhauru) Both modes broken because of
+        # https://github.com/EnzymeAD/Enzyme.jl/issues/2035
```
Comment (wsmoses) [inline test/integration/Bijectors/runtests.jl:150]: So this issue is essentially due to Enzyme 0.13 not being supported by the custom rule on bijectors on all uses of roots.jl to define rules.

x/ref: https://github.com/TuringLang/Bijectors.jl/issues/339

I think this needs to be fixed in bijectors.jl before we can take this in here

File: test/integration/Bijectors/runtests.jl (hunk 2 above)
Comment (mhauru) [reply test/integration/Bijectors/runtests.jl]: Done.

File: test/integration/Bijectors/runtests.jl (hunk 3 above)
Comment (mhauru) [reply test/integration/Bijectors/runtests.jl:148]: I would propose merging this as is while we fix this, to get the vast majority of the tests in this PR into CI. Bijectors is pinned to a specific version, so once the necessary changes have been made in Bijectors we can release a new version of IT, and then make a PR that marks this unbroken as it bumps the Bijectors version of these tests.

**Comment by mhauru** [comment]
Tests pass and nothing is marked as `skip`ped any more. Happy to merge @wsmoses?

**Pull request merged**